### PR TITLE
Add Eslint config and fix files for rules

### LINKED ISSUE #160: Add ESLint for Code Linting + Enforcing Syntax
[ESLint](https://eslint.org/) helps us set rules for syntax in our code. Examples include enforving the use of single quotes, use camelcase for variable naming and more. Use [this guide](https://eslint.org/docs/user-guide/getting-started) to get started by adding it to the repository.

- If it asks, use a eslint file with JSON format
- The list of rules we can set are [here](https://eslint.org/docs/rules/)
- [example eslint JSON file](https://github.com/evanugarte/SmartPhotoCategorizer/blob/master/.eslintrc.json)

### PR DESCRIPTION
**Pull request by realCalvin**
Resolves #160

### CODE REVIEW

**Comment by realCalvin** [comment]
@evanugarte  help

How should I resolve this ESLint problem aside from using eslint ignore?
image

Also some other questions.
1) How should I deal with code that console logs errors when performing an action? (might be useful for troubleshooting in future if something breaks, example below) I'm currently just making ESLint ignore the line.
image

2) Dealing with camel case variables that (I believe) involves prenamed variables in the API. (mailChimp.js) I'm currently just making ESLint ignore the line.
image

3) ESLint isn't giving any warnings for missing semicolons. (unless this rule was removed, idk).

**Comment by realCalvin** [comment]
@ThaiQ can you review. thank you :D

**Comment by evanugarte** [comment]
@realCalvin Answers to your questions:

1. you can allow `console.error` if you want. refer to [this rule](https://github.com/evanugarte/SmartPhotoCategorizer/blob/1756719c9f7c1f4904af3735740ae8e1fe75744a/.eslintrc.json#L47)
1. we deleted mailchimp lol
1. is this still an issue?

**Comment by realCalvin** [comment]
> @realCalvin Answers to your questions:
> 
> 1. you can allow `console.error` if you want. refer to [this rule](https://github.com/evanugarte/SmartPhotoCategorizer/blob/1756719c9f7c1f4904af3735740ae8e1fe75744a/.eslintrc.json#L47)
> 2. we deleted mailchimp lol
> 3. is this still an issue?

It should all be working. For semicolons, turns out it was automatically added sooo

**Comment by realCalvin** [comment]
Added console logging for errors and fixed bugs I made when doing eslint stuff.

File: api/mailer/mailer.js [hunk 1]
```diff
@@ -107,32 +107,29 @@ const send = ({ templateType, recipientEmail, recipientName }) => {
...
-          else callback(JSON.parse(token))
-        })
+          // This needs a .then or async statement so it doesnt 
+          // move until a new access token is created
+          if (checkIfTokenIsExpired(JSON.parse(token))) refreshAccessToken();
+          else callback(JSON.parse(token));
+        });
       }
-    )
-  }).catch(error => console.log(error))
```
[the remark this answers is not in the document]
Comment (realCalvin) [reply api/mailer/mailer.js]: I get an error

image
image

File: api/routes/3DPrintingForm.js [hunk 2]
```diff
@@ -1,21 +1,21 @@
-'use strict'
+'use strict';
```
[the remark this answers is not in the document]
Comment (realCalvin) [reply api/routes/3DPrintingForm.js]: removed

File: src/APIFunctions/Image.js [hunk 3]
```diff
@@ -6,9 +6,9 @@
  */
 export async function validateImageURL (url) {
   return new Promise((resolve, reject) => {
-    let img = new Image()
-    img.onload = () => resolve(img.height)
-    img.onerror = reject
-    img.src = url
-  })
+    const img = new Image();
```
[the remark this answers is not in the document]
Comment (realCalvin) [reply src/APIFunctions/Image.js]: changed

File: api/mailer/mailer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (evanugarte) [reply api/mailer/mailer.js]: there shouldnt be a reject there.

[the remark this answers is not in the document]
Comment (realCalvin) [reply api/mailer/mailer.js]: Fixed!

**Comment by realCalvin** [comment]
Fixed all the ESLint errors. Checked the website to make sure everything works. One thing that is broken is the Profile page, but it seems that someone else modified it on the dev branch. (broken on dev as well)

**Comment by realCalvin** [comment]
> Can you get rid off these files in your api/ directory?
> image

oops fixed

**Pull request merged**